llvm/include/llvm/Object/DXContainer.h

```diff
@@ -149,6 +152,24 @@ struct RootConstantView : RootParameterView {
   }
 };
 
+struct RootDescriptorView : RootParameterView {
+  static bool classof(const RootParameterView *V) {
+    return (V->Header.ParameterType ==
+                llvm::to_underlying(dxbc::RootParameterType::CBV) ||
+            V->Header.ParameterType ==
+                llvm::to_underlying(dxbc::RootParameterType::SRV) ||
+            V->Header.ParameterType ==
+                llvm::to_underlying(dxbc::RootParameterType::UAV));
+  }
+
+  llvm::Expected<dxbc::RST0::v1::RootDescriptor> read(uint32_t Version) {
+    if (Version == 1)
```

Comment:
```suggestion
    if (Version == 0)
```
A little concerning that this passes the testcases because IIUC this is reading the opposite version of what would be expected

Reply:
I've used 1 and 2 instead of 0 and 1 to represent each version.

Reply:
Ah right, okay then this LGTM